File: 448_Find_All_Numbers_Disappeared_in_an_Array/find_disappeared_numbers.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        indices = set()
        values = set()
        for i, v in enumerate(nums):
            indices.add(i+1)
            values.add(v)
        return list(indices - values)
        

File: 448_Find_All_Numbers_Disappeared_in_an_Array/find_disappeared_numbers.java
public class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> missingNum = new ArrayList<Integer>();
        Set indices = new HashSet();
        Set values = new HashSet();
        
        for (int i = 0; i < nums.length; i++) {
            indices.add(i+1);
            values.add(nums[i]);
        }
        
        indices.removeAll(values);
        missingNum.addAll(indices);
        return missingNum;
    }
}
